### ISSUE TITLE
catkin_make throws error.

### ISSUE THREAD

**Issue by paulsammut** [stonier/qt_ros#22]
Just tried to build an empty test package and catkin_make failed.

ROS Version: Kinetic
OS: Ubuntu 16.04
qt_ros version (number found in the package.xml): 0.2.9

Steps to produce problem:

`cd ~/catkin_ws/src`
`catkin_create_qt_pkg test_qt`
`cd ~/catkin_ws && catkin_make`

Error output from catkin_make:

```
usr/include/boost/type_traits/detail/has_binary_operator.hp:50: Parse error at "BOOST_JOIN"
test_qt/CMakeFiles/test_qt.dir/build.make:72: recipe for target 'test_qt/include/test_qt/moc_qnode.cxx' failed
make[2]: *** [test_qt/include/test_qt/moc_qnode.cxx] Error 1
make[2]: *** Waiting for unfinished jobs....
usr/include/boost/type_traits/detail/has_binary_operator.hp:50: Parse error at "BOOST_JOIN"
test_qt/CMakeFiles/test_qt.dir/build.make:77: recipe for target 'test_qt/include/test_qt/moc_main_window.cxx' failed
make[2]: *** [test_qt/include/test_qt/moc_main_window.cxx] Error 1
CMakeFiles/Makefile2:3131: recipe for target 'test_qt/CMakeFiles/test_qt.dir/all' failed
make[1]: *** [test_qt/CMakeFiles/test_qt.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
Invoking "make -j8 -l8" failed

```

**Comment by paulsammut** [comment]
I was able to get it to compile by editing the following for my package:

**file: my_pacakge/include/test_qt/qnode.hpp**
Encapsulate includes with the following #ifndef block:

```
#ifndef Q_MOC_RUN
#include <ros/ros.h>
#include <string>
#include <QThread>
#include <QStringListModel>
#endif
```

I found this information here: https://answers.ros.org/question/233786/parse-error-at-boost_join/

No idea why/how this makes it work.

**Comment by stonier** [comment]
Duplicate, just closed and released 0.2.10 yesterday in #19.

**Comment by paulsammut** [comment]
Thanks you!
